internal/scheduler/submitcheck.go

```diff
@@ -134,11 +135,35 @@ func (srv *SubmitChecker) updateExecutors(ctx *armadacontext.Context) {
 	srv.jobSchedulingResultsCache.Purge()
 }
 
-func (srv *SubmitChecker) CheckApiJobs(jobs []*api.Job) (bool, string) {
+func (srv *SubmitChecker) CheckJobDbJobs(jobs []*jobdb.Job) (bool, string) {
 	return srv.check(schedulercontext.JobSchedulingContextsFromJobs(srv.priorityClasses, jobs))
 }
 
-func (srv *SubmitChecker) CheckJobDbJobs(jobs []*jobdb.Job) (bool, string) {
+func (srv *SubmitChecker) CheckApiJobs(es *armadaevents.EventSequence) (bool, string) {
```

Comment:
This is a;l a bit messy, but I think we should refactor the way that the submitCheck works ina subsequent PR